crates/sui-types/src/messages.rs

```diff
@@ -55,12 +55,15 @@ pub enum CallArg {
     ObjVec(Vec<ObjectArg>),
 }
 
-#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy, Serialize, Deserialize)]
 pub enum ObjectArg {
     // A Move object, either immutable, or owned mutable.
     ImmOrOwnedObject(ObjectRef),
     // A Move object that's shared and mutable.
-    SharedObject(ObjectID),
+    SharedObject {
+        id: ObjectID,
+        shared_at_version: SequenceNumber,
```

Comment:
I think `initial_version` would be less confusing, but its not a big deal.

Reply:
The issue potentially with `initial_version` is that it is not the initial version of the object, but just the one it was shared. If you do not think it will be confusing, we can change it, sure.

Reply:
Had the same thought as Todd here about wanting to distinguish between the first version is shared at, vs the initial version it was created with.  `initial_shared_version` is also another option?  I will wait to get final thoughts before making a change (or not).

Reply:
`initial_shared_version` might be better. Don't let me name things

Reply:
> does this mean the user always needs to pass in this version explicitly going forward? it seems quite tedious

It will be required, yes. But it should be very easy to determine, as it is present in the `Owner` field of the object. So it will be similar to how we have to populate `ObjectRef` now for non-shared objects, in that you will have to populate the arg with some extra information (other than the ID) that can be easily fetched automatically

Reply:
Now renamed to `initial_shared_version`!